ops/e2e.Dockerfile

```diff
@@ -0,0 +1,224 @@
+# Base stage for all other Triton-based build and runtime stages
+###########################################################################################
+FROM ${BASE_IMAGE} as cuda_base
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PATH /opt/tritonserver/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:${PATH}
+ENV LD_LIBRARY_PATH /opt/tritonserver/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
+
+# Env variables users may expect from typical CUDA containers
+ENV CUDA_VERSION 11.2.1
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+ENV NVIDIA_REQUIRE_CUDA "cuda>=11.2 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=440,driver<441 driver>=450"
+ENV CUDNN_VERSION 8.2.0.53
+
+# TODO:
+#    BOTH:
+#        * cuda-libraries-11-2
+#    RUNTIME:
+#    BUILD:
+#        * cuda-compiler-11-2
+#        * cuda-libraries-dev-11-2
+#        * cuda-nvml-dev-11-2
+
+########################
+# Install CUDA and cuDNN
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+    gnupg2 \
+    curl \
+    ca-certificates \
+ && curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/7fa2af80.pub \
+    | apt-key add - \
+ && echo "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64 /" \
+    > /etc/apt/sources.list.d/cuda.list \
+ && echo "deb https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64 /" \
+    > /etc/apt/sources.list.d/nvidia-ml.list \
+ && apt-get update \
+ && apt-get install -y --no-install-recommends \
+    cuda-cudart-11-2=11.2.146-1 \
+    cuda-compat-11-2 \
+ && apt-get install -y \
+    cuda-libraries-11-2 \
+ && ln -s cuda-11.2 /usr/local/cuda \
+ && apt-get install -y --no-install-recommends \
+    libcudnn8=$CUDNN_VERSION-1+cuda11.3 \
+    libcudnn8-dev=$CUDNN_VERSION-1+cuda11.3 \
+ && apt-mark hold libcudnn8 \
+ && rm -rf /var/lib/apt/lists/*
+
+FROM ${CUDA_IMAGE} AS base
+##################
+# Install TensorRT
+
+# NOTE: This file must be downloaded to the ops/stage directory by following
+# the directions at
+# https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#downloading
+COPY ops/stage/nv-tensorrt-repo-ubuntu2004-cuda11.3-trt8.0.0.3-ea-20210423_1-1_amd64.deb /tmp/tensorrt.deb
+
+RUN dpkg -i /tmp/tensorrt.deb \
+ && apt-key add /var/nv-tensorrt-repo-ubuntu2004-cuda11.3-trt8.0.0.3-ea-20210423/7fa2af80.pub \
+ && apt-get update \
+ && apt-get install -y \
+    docker.io \
+    libb64-dev \
+    libnvinfer-dev \
+    libnvinfer-plugin-dev \
+    libnvparsers-dev \
+    libnvonnxparsers-dev \
+    libnvinfer-samples \
+    libre2-dev \
+    libtool \
+    onnx-graphsurgeon \
+    pkg-config \
+    python3-dev \
+    python3-libnvinfer-dev \
+    python3-opencv \
+    python3-pip \
+    python3-setuptools \
+    software-properties-common \
+    tensorrt \
+ && rm /tmp/tensorrt.deb \
+ && rm -rf /var/lib/apt/lists/*
```

Comment:
I think you'll want to run `apt clean` at the end of each command chain, that will delete all cached packages as well.